container/container.go

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Genome Research Ltd.
+ *
+ * Author: Sendu Bala <sb10@sanger.ac.uk>, Ashwini Chhipa <ac55@sanger.ac.uk>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ ******************************************************************************/
+
+package container
+
+// this file has functions for dealing with containers
+
+import (
+	"context"
+	"strings"
+)
+
+// Container struct represents a container type with specific properties.
+type Container struct {
+	ID     string
+	Names  []string
+	client Interactor
+}
+
+// Stats returns the current memory usage (RSS, in MB) and total CPU usage (in
+// seconds) of this container.
+func (c *Container) Stats(ctx context.Context) (*Stats, error) {
+	stats, err := c.client.ContainerStats(ctx, c.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return stats, nil
+}
+
+// TrimPrefixName trims the '/' from this container's names.
```

Comment:
I think this makes more sense if named `TrimNamePrefixes`.

Reply:
Also document that "You must call this after adding names that might contain the prefix."

Reply:
Renamed and updated the doc.